File: betabot_square_away/src/betabot_move_custom_service_client.py
#!/usr/bin/env python

from move_betabot_in_square_custom.srv import *
import rospy
import sys
from geometry_msgs.msg import Twist


# Wait for service client /gazebo/* to be running
def move_client(side, repetitions):
    rospy.wait_for_service('/move')
    try:
            move = rospy.ServiceProxy('/move', BetabotCustomServiceMessage)
            resp1 = move(side, repetitions)
            return resp1.success

    except rospy.ServiceException:
            print "Service call failed"

def usage():
    return "%s [side repetitions]"%sys.argv[0]

if __name__ == "__main__":
    if len(sys.argv) == 3:
        side = float(sys.argv[1])
        repetitions = int(sys.argv[2])
    else:
        print usage()
        sys.exit(1)
    print "Requesting movement"
    print "Successful?? %s"%(move_client(side,repetitions))
    

File: betabot_square_away/src/betabot_move_custom_service_client.cpp
#include "ros/ros.h"
#include "betabot_square_away/BetabotCustomServiceMessage.h"
#include <cstdlib>

int main(int argc, char **argv)                // Node Main Function
{
  ros::init(argc, argv, "move_client");   // Initializes Node Name
  if(argc != 3)
  {
    ROS_INFO("usage: add_two_ints_client X Y");
    return 1;
  }
  ros::NodeHandle nh;                          // Node handle declaration

  // Hook our service clients to the master
  ros::ServiceClient client = nh.serviceClient<betabot_square_away::BetabotCustomServiceMessage>("move");
  
  // create request and response 
  betabot_square_away::BetabotCustomServiceMessage srv;

  srv.request.side = atoll(argv[1]);
  srv.request.repetitions = atoll(argv[2]);
  if (client.call(srv))
  {
      if(srv.response.success){
        ROS_INFO("Successful move");
      }
      else
      {
        ROS_INFO("Failed to move right");
      }
  }
  else
  {
    ROS_ERROR("Failed to call service move");
    return 1;
  }
  return 0;
}